docs/platform-integration/storage/secure-storage.md

```diff
@@ -0,0 +1,151 @@
+    ...
+    <application android:allowBackup="false" ... >
+        ...
+    </application>
+</manifest>
+```
+
+### Selective backup
+
+Auto Backup can be configured to disable specific content from backing up. You can create a custom rule set to exclude `SecureStore` items from being backed up.
+
+01. Set the `android:fullBackupContent` attribute in your _AndroidManifest.xml_:
+
+    ```xml
+    <application ...
+        android:fullBackupContent="@xml/auto_backup_rules">
+    </application>
+    ```
+
+01. Create a new XML file named _auto_backup_rules.xml_ in the _Resources/xml_ directory with the build action of **AndroidResource**. Set the following content that includes all shared preferences except for `SecureStorage`:
+
+    ```xml
+    <?xml version="1.0" encoding="utf-8"?>
+    <full-backup-content>
+        <include domain="sharedpref" path="."/>
+        <exclude domain="sharedpref" path="${applicationId}.mauiessentials.xml"/>
+    </full-backup-content>
+    ```
+
+# [iOS](#tab/ios)
+
+When developing on the **iOS simulator**, enable the **Keychain** entitlement and add a keychain access group for the application's bundle identifier.
+
+Open the _Entitlements.plist_ in the project and find the **Keychain** entitlement and enable it. This will automatically add the application's identifier as a group.
+
+In the project properties, under **iOS Bundle Signing** set the **Custom Entitlements** to **Entitlements.plist**.
+
+> [!TIP]
+> When deploying to an iOS device, this entitlement isn't required and should be removed.
+
+# [Windows](#tab/windows)
+
+No setup is required.
+
+-----
+<!-- markdownlint-enable MD025 -->
+
+## Use secure storage
+
+The following code examples demonstrate how to use secure storage.
+
+> [!TIP]
+> It's possible that an exception is thrown when calling `GetAsync` or `SetAsync`. This can be caused by a device not supporting secure storage, encryption keys changing, or corruption of data. it's best to handle this by removing and adding the setting back if possible.
+
+### Write a value
+
+To save a value for a given _key_ in secure storage:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_set":::
+
+### Read a value
+
+To retrieve a value from secure storage:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_get":::
+
+> [!TIP]
+> If there isn't a value associated with the key, `GetAsync` returns `null`.
+
+### Remove a value
+
+To remove a specific value, remove the key:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_remove":::
+
+To remove all values, use the `RemoveAll` method:
+
+:::code language="csharp" source="../snippets/shared_1/Storage.cs" id="secstorage_remove_all":::
+
+## Platform differences
+
+This section describes the platform-specific differences with the secure storage API.
+
+<!-- markdownlint-disable MD025 -->
+<!-- markdownlint-disable MD024 -->
+# [Android](#tab/android)
```

Comment:
I vaguely recall that there may have been a change to the APIs used here on Android. Maybe just add a todo to investigate?

Reply:
Yes. we removed all the old stuff for older platforms so now it is only 23+

Reply:
@jamesmontemagno are you saying that this feature is only supported on 23+?

Reply:
I think this section (Android platform differences) needs an update, based on this PR - https://github.com/dotnet/maui/pull/2807

@jamesmontemagno Can you comment on what the precise updates should be?

At an absolute minimum, the link to shared preferences should go because it's stored as `EncryptedSharedPreferences`.

Reply:
Yep! This section should be mostly entirely rewritten.
We no longer use md5 and KeyStore as they're outdated. Instead, we're using `EncryptedSharedPreferences`

`SharedPreferences` is still kinda relevant though because `EncryptedSharedPreferences` is technically a implementation of `SharedPreferences`

Here are some new relevant links that can be included in this section:
https://developer.android.com/topic/security/data
https://developer.android.com/reference/androidx/security/crypto/EncryptedSharedPreferences

Reply:
I believe I represented the information correct. Please see this commit to understand the changes I just made: https://github.com/dotnet/docs-maui/pull/536/commits/91f0997d3c3650c626f41d70e0fa5d8271b963af

@davidbritch feel free to modify the content here as you see if and then approve/squash this on Monday.